crates/fuel-core/src/query/message.rs

```diff
@@ -266,20 +261,21 @@ fn message_receipts_proof<T: MessageProofData + ?Sized>(
         tree.push(id.as_ref());
     }
 
-    // If we found the leaf proof index then return the proof.
-    match proof_index {
-        Some(proof_index) => {
-            // Generate the actual merkle proof.
-            match tree.prove(proof_index) {
-                Some((_, proof_set)) => Ok(Some(MerkleProof {
-                    proof_set,
-                    proof_index,
-                })),
-                None => Ok(None),
-            }
-        }
-        None => Ok(None),
-    }
+    // Check if we found a leaf.
+    let Some(proof_index) = proof_index else {
+        return Err(anyhow::anyhow!("unable to find leaf").into())
+    };
+
+    // Get the proof set.
+    let Some((_, proof_set)) = tree.prove(proof_index) else {
+        return Err(anyhow::anyhow!("unable to get proof set").into());
```

Comment:
```suggestion
        return Err(anyhow::anyhow!("Unable to generate the Merkle proof for the message from its receipts").into());
```

Reply:
Updated in https://github.com/FuelLabs/fuel-core/pull/2258/commits/2199e8760d9eb9f72650fe36d8eefa26162008d6